### PR TITLE
[Communication] - PhoneNumbers - Hide unused types and making operations testable

### PR DESCRIPTION
**Pull request by RezaJooyandeh**

### CODE REVIEW

**Review by pavelprystinka — APPROVED** [review]

**Review by beltr0n — APPROVED** [review]
Assuming this doesn't need a Changelog update?

**Comment by RezaJooyandeh** [comment]
@beltr0n good catch. I pushed a new update and updated the changelog

**Review by beltr0n — APPROVED** [review]

**Review by sacheu — APPROVED** [review]

File: sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md [hunk 1]
```diff
@@ -10,8 +10,9 @@
   - GetPhoneNumbers -> GetPurchasedPhoneNumbers.
   - GetPhoneNumbersAsync -> GetPurchasedPhoneNumbersAsync.
 - PhoneNumbersModelFactory static method AcquiredPhoneNumber is renamed to PurchasedPhoneNumber.
-- PurchasePhoneNumbersOperation extends Operation instead of Operation<Response>.
-- ReleasePhoneNumberOperation extends Operation instead of Operation<Response>.
+- PurchasePhoneNumbersOperation and ReleasePhoneNumberOperation extend Operation instead of Operation<Response>.
+- Added protected constructor to PurchasePhoneNumbersOperation and ReleasePhoneNumberOperation for mocking.
```
Comment (pavelprystinka) [inline sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md:14]: Should that be added to 'Added' section instead of braking changes?

Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.PhoneNumbers/CHANGELOG.md]: Updated it

**Pull request merged**